parrot/recorder.go

```diff
@@ -0,0 +1,142 @@
+package parrot
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"time"
+)
+
+type Recorder struct {
+	URL    string `json:"url"`
+	server *http.Server
+
+	recordChan chan *RouteCall
+	errChan    chan error
+}
+
+// RouteCall records that a route was called
+type RouteCall struct {
+	RouteID  string             `json:"route_id"`
+	Request  *RouteCallRequest  `json:"request"`
+	Response *RouteCallResponse `json:"response"`
+}
+
+// RouteCallRequest records the request made to a route
+type RouteCallRequest struct {
+	Method string      `json:"method"`
+	URL    *url.URL    `json:"url"`
+	Header http.Header `json:"header"`
+	Body   []byte      `json:"body"`
+}
+
+// RouteCallResponse records the response from a route
+type RouteCallResponse struct {
+	StatusCode int         `json:"status_code"`
+	Header     http.Header `json:"header"`
+	Body       []byte      `json:"body"`
+}
+
+func NewRecorder() (*Recorder, error) {
+	r := &Recorder{
+		recordChan: make(chan *RouteCall),
+		errChan:    make(chan error),
+	}
+
+	// TODO: Will need a way to send out the URL to an external service (e.g. Parrotserver running in a docker container)
```

Comment:
Here is our guide for components: https://smartcontractkit.github.io/chainlink-testing-framework/developing/developing_components.html
It can solve this problem, return it as an `*Output* as other components do, and, ideally, move it under "components" dir, wdyt?

Reply:
That works fine for me. Will look at moving it.